test/tasks/demo.test.js

```diff
@@ -38,112 +38,164 @@ describe('Demo task', function() {
 				});
 		});
 
-		it('should not error with a custom config file', function(done) {
-			fs.writeFileSync('bower.json', '{"name":"o-test"}', 'utf8');
-			fs.copySync('demos/src/config.json', 'demos/src/mysupercoolconfig.json');
+		it('should build demos defined in custom configuration files when specified', function (done) {
+			// Define template and custom config file
+			const testKey = 'custom-config';
+			const templatePath = `demos/src/${testKey}.mustache`;
+			const templateMarkup = `<div>${testKey}</div>`;
+			const configPath = `demos/src/${testKey}.json`;
+			const config = {
+				options: {
+					sass: 'demos/src/demo.scss',
+					js: 'demos/src/demo.js',
+					bodyClasses: testKey
+				},
+				demos: [
+					{
+						name: testKey,
+						description: 'A test demo.',
+						template: templatePath
+					}
+				]
+			};
+			// Write templates and config to the test directory
+			fs.writeFileSync(configPath, JSON.stringify(config), 'utf8');
+			fs.writeFileSync(templatePath, templateMarkup, 'utf8');
+			// Assert the template defined in custom config is built without error.
+			demo(gulp, { demoConfig: configPath })
+				.on('data', function (file) {
+					proclaim.include(file.path, `${testKey}.html`);
+					proclaim.include(file.contents.toString('utf8'), templateMarkup);
+					done();
+				}).on('error', function errorHandler(err) {
+					throw new Error('There was an error: ' + err.message);
+				});
+		});
+
+		it('should build demos defined in origami.json', function(done) {
+			const createdFiles = [];
 			const demoStream = demo(gulp, {
-				demoConfig: 'demos/src/mysupercoolconfig.json'
-			})
-			.on('error', function errorHandler(err) {
-				// It will throw a template not found error which is fixed in "should build html" test
-				proclaim.notEqual(err.message, 'Couldn\'t find demos config path, checked: demos/src/mysupercoolconfigs.json');
-				fs.unlink('demos/src/mysupercoolconfig.json');
-				demoStream.removeListener('error', errorHandler);
+				demoConfig: 'origami.json'
+			}).on('data', function (file) {
+				createdFiles.push({
+					path: file.path,
+					content: file.contents.toString('utf8'),
+				});
+			}).on('error', function(err) {
+				throw new Error('There was an error: ' + err.message);
```

Comment:
The Mocha docs mention for asynchronous tests using the `done` callback need to pass the error as the first argument to `done` to register it correctly as a test failure -- https://mochajs.org/#asynchronous-code

```suggestion
				done(new Error('There was an error: ' + err.message));
```